kubetest2/internal/deployers/eksapi/infra.go

```diff
@@ -80,95 +82,210 @@ func (i *Infrastructure) subnets() []string {
+	describeInput := &ec2.DescribeSecurityGroupsInput{
+		Filters: []ec2types.Filter{
 			{
-				ParameterKey:   aws.String("Subnet01AZ"),
-				ParameterValue: aws.String(subnetAzs[0]),
+				Name:   aws.String("group-name"),
+				Values: []string{groupName},
 			},
 			{
-				ParameterKey:   aws.String("Subnet02AZ"),
-				ParameterValue: aws.String(subnetAzs[1]),
+				Name:   aws.String("vpc-id"),
+				Values: []string{vpcId},
 			},
 		},
 	}
-	if opts.ClusterRoleServicePrincipal != "" {
-		input.Parameters = append(input.Parameters, cloudformationtypes.Parameter{
-			ParameterKey:   aws.String("AdditionalClusterRoleServicePrincipal"),
-			ParameterValue: aws.String(opts.ClusterRoleServicePrincipal),
-		})
+
+	resp, err := m.clients.EC2().DescribeSecurityGroups(context.TODO(), describeInput)
+	if err != nil {
+		return "", fmt.Errorf("failed to describe security groups, %v", err)
 	}
-	if opts.EKSEndpointURL != "" {
-		input.Tags = []cloudformationtypes.Tag{
-			{
-				Key:   aws.String(eksEndpointURLTag),
-				Value: aws.String(opts.EKSEndpointURL),
-			},
-		}
+
+	if len(resp.SecurityGroups) > 0 {
+		return *resp.SecurityGroups[0].GroupId, nil
+	}
+
+	sgInput := &ec2.CreateSecurityGroupInput{
+		GroupName:   aws.String(groupName),
+		Description: aws.String("SSH access for worker nodes"),
+		VpcId:       aws.String(vpcId),
 	}
-	klog.Infof("creating infrastructure stack...")
-	out, err := m.clients.CFN().CreateStack(context.TODO(), &input)
+
+	sgOutput, err := m.clients.EC2().CreateSecurityGroup(context.TODO(), sgInput)
 	if err != nil {
-		return nil, err
+		return "", fmt.Errorf("failed to create security group, %v", err)
 	}
-	klog.Infof("waiting for infrastructure stack to be created: %s", *out.StackId)
-	err = cloudformation.NewStackCreateCompleteWaiter(m.clients.CFN()).
-		Wait(context.TODO(),
-			&cloudformation.DescribeStacksInput{
-				StackName: out.StackId,
+
+	ingressInput := &ec2.AuthorizeSecurityGroupIngressInput{
+		GroupId: aws.String(*sgOutput.GroupId),
+		IpPermissions: []ec2types.IpPermission{
+			{
+				IpProtocol: aws.String("tcp"),
+				FromPort:   aws.Int32(22),
+				ToPort:     aws.Int32(22),
+				IpRanges: []ec2types.IpRange{
+					{
+						CidrIp: aws.String("0.0.0.0/0"),
+					},
+				},
 			},
-			infraStackCreationTimeout)
+		},
+	}
+
+	_, err = m.clients.EC2().AuthorizeSecurityGroupIngress(context.TODO(), ingressInput)
 	if err != nil {
-		return nil, fmt.Errorf("failed to wait for infrastructure stack creation: %w", err)
+		return "", fmt.Errorf("failed to authorize ingress, %v", err)
 	}
-	klog.Infof("getting infrastructure stack resources: %s", *out.StackId)
-	infra, err := m.getInfrastructureStackResources()
+
+	return *sgOutput.GroupId, nil
+}
+func (m *InfrastructureManager) getOrCreateSSHKeyPair(publicKeyMaterial string) (string, error) {
+	keyPairName := "StaticKubetest2SSHKeyPair"
+	describeOutput, err := m.clients.EC2().DescribeKeyPairs(context.TODO(), &ec2.DescribeKeyPairsInput{
```

Comment:
would you use `"SSHKeyPair"`? https://github.com/aws/aws-k8s-tester/blob/652d01a5bacbcef8d9d0e42ce1ae4abe88044d99/kubetest2/internal/deployers/eksapi/nodegroup.go#L224

Reply:
the SSHKeyPair was created by the infra, which is ephemeral, so have to create a static one separately